aiocouch/remote.py

```diff
@@ -100,8 +100,14 @@ def __init__(self, server, user=None, password=None, cookie=None, **kwargs):
         return await self._get("/_all_dbs", params)
 
     async def close(self):
+        # If ClientSession has TLS/SSL connections, it is needed to wait 250 ms
+        # before closing, see https://github.com/aio-libs/aiohttp/issues/1925.
+        has_ssl_conn = self._http_session.connector and any(
+            any(hasattr(handler.transport, '_ssl_protocol') for handler, _ in conn)
+            for conn in self._http_session.connector._conns.values())
         await self._http_session.close()
-        await asyncio.sleep(0.250)
+        if has_ssl_conn:
+            await asyncio.sleep(0.250)
```

Comment:
According to the aiohttp documentation, you still need a sleep of 0 if SSL isn't used:

> For a ClientSession without SSL, a simple zero-sleep (await asyncio.sleep(0)) will suffice:

Reply:
Ah! Thanks. Updated:
```diff
@@ -106,8 +106,7 @@ class RemoteServer(object):
             any(hasattr(handler.transport, '_ssl_protocol') for handler, _ in conn)
             for conn in self._http_session.connector._conns.values())
         await self._http_session.close()
-        if has_ssl_conn:
-            await asyncio.sleep(0.250)
+        await asyncio.sleep(0.250 if has_ssl_conn else 0)
 
     @raises(401, "Authentification failed, check provided credentials.")
     async def _check_session(self):
```